### DISCUSSION TITLE
Command line option to specify inventory file

### DISCUSSION THREAD

**Discussion by darkdragon-001** [ansible/ansible-lint#2024] [Ideas]
How can I specify a custom inventory file? `-i` seems to be not supported as with `ansible-playbook`.

**Comment by ssbarnea** [comment]
There are no plans to do this, for two reasons:

- you can still specify inventory using `ANSIBLE_INVENTORY` or `ansible.cfg`
- any playbook or role should be lintable even without an inventory. If you get errors, it means that you did not use proper syntax to avoid errors when inventory lacks those groups, hosts or variables.

Basically, if you reach the conclusion that you need an inventory, is very likely that you have another problem, like add default values. Make use of `fail` to stop execution when prerequisites are missing (like mandatory variables).
